New Bank/NewBank/newbank/server/NewBank.java

```diff
@@ -46,13 +47,30 @@ public synchronized CustomerID checkLogInDetails(String userName, String passwor
 		return null;
 	}
 
-	// commands from the NewBank customer are processed in this method
+	// commands from the NewBank cusbtomer are processed in this method
 	public synchronized String processRequest(CustomerID customer, String request){
 		if(customers.containsKey(customer.getKey())) {
 			String[] input = request.split(" "); // create an array of the parsed input string
 			if (request.startsWith("NEWACCOUNT")){
 				return openAccount(customer, request.substring(request.indexOf(" ") + 1)); // +1 to remove the leading space
 			}
+			if (request.startsWith("REGISTERLENDER")){
+				if(input.length < 1) { // return fail if not enough information is provided
+					return "FAIL";
+				}
+				Customer cust = customers.get(customer.getKey());
+				return registerLender(cust, input[1]);
+			}
+			if (request.startsWith("BORROWMICROLOAN")){
+				if(input.length < 4) { // return fail if not enough information is provided
+					return "FAIL";
+				}
+				Customer cust = customers.get(customer.getKey());
+				return borrowMicroLoan(cust, input[1], input[2], input[3], input[4]);
+			}
+			if (request.startsWith("SHOWMICROLENDERS")){
+				return showMicroLenders(); // +1 to remove the leading space
+			}
```

Comment:
SHOWMICROLENDERS feature could be added to the protocol.txt

Reply:
Done